tools/codegen/gen.py

```diff
@@ -176,6 +176,94 @@ def __call__(self, f: NativeFunction) -> Optional[str]:
         return f'm.def({cpp_string(str(f.func))});\n'
 
 
+def _num_leading_spaces(line: str) -> int:
+    return len(line) - len(line.lstrip())
+
+
+# Unindents all lines in code. Each line gets unindented the same amount;
+# that amount is equal to the smallest number of leading spaces across all lines
+def deindent(code: str) -> str:
+    lines = code.split('\n')
+    min_leading_spaces = min(map(_num_leading_spaces, lines))
+    lines = [line[min_leading_spaces:] for line in lines]
+    return '\n'.join(lines)
+
+
+# Generates UnambiguousTorchOperators.h and UnambiguousTorchOperators.cpp.
+# These provide macros that, given an operator and overload name, allow users
+# to access an "un-overloaded" function version of the operator. This
+# is useful for extension writers who want to (1) want to decltype the operator
+# and (2) don't want to worry about method-only operators.
+@dataclass(frozen=True)
+class UnambiguousTorchOperators:
+    target: Union[
+        Literal[Target.DECLARATION],
+        Literal[Target.DEFINITION]
+    ]
+
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> Optional[str]:
+        unambiguous_name = self.unambiguous_function_name(f)
+        # NB: requires_grad is the only exception to the rule because
+        # its const correctness is questionable.
+        if unambiguous_name in set(['requires_grad_']):
```

Comment:
most of the other parsing in the codegen works directly off of the NativeFunction object instead of some other converted value. So you don't really need to call `self.unambiguous_function_name()` to do the comparison here; you can just do `if str(f.func.name) in set(['requires_grad_'])`